### PR TITLE
Remove tslint in favour of eslint

### PR DESCRIPTION
**Pull request by ematipico**
**Do you want to request a *feature* or report a *bug*?**
Feature, better a chore.

**What is the current behavior?**
At the moment we use two linters to lint the projects

**If the current behavior is a bug, please provide the steps to reproduce.**
N/A

**If this is a feature request, what is motivation or use case for changing the behavior?**
Using eslint to lint all the files.

**Main changes that there are in this PR:**
- created a super generic `.eslintrc.js` file. Which we already have and I removed all the the rules regarding `eslint-plugin-node`.
- inside the `cli` folder I created a new eslint config file that inherits from the main one and it adds all the node rules that have been removed from the main config file
- inside each package I crated an eslint file that takes care of using the proper typescript parser plus and plugin. I also added the prettier configuration so it won't collide with formatting
- tests should be always inside a folder so we can have a specific eslint config file for those: by default typescript-eslint wants for every function to return the type (which is good), but when we define the tests, we always have `describe("", () => {})` which is annoying. So better disable the rule for test files.

### CODE REVIEW

**Comment by pranshuchittora** [comment]
IMO both ESLint & TSLint have some voids. Pls read [this](https://medium.com/palantir/tslint-in-2019-1a144c2317a9)

**Comment by ematipico** [comment]
Yeah I know but it's difficult to maintain two linters with two different way of configuring things. Now that eslint supports TypeScript and core members of TypeScript are helping `@typescript-eslint` I'd say it's a good thing to move to just one to help and make maintainability easier.

**Comment by anshumanv** [comment]
> In order to avoid bifurcating the linting tool space for TypeScript, we therefore plan to deprecate TSLint and focus our efforts instead on improving ESLint’s TypeScript support. 

Positive change IMO. ✨

**Comment by ematipico** [comment]
TypeScript interfaces are a bit messed up it's a full of duplicates. 
After this PR we should think about doing the following things: 
- remove our types about JSCodeshift and use `@types/jscodeshift`
- remove out yeoman types and use `@types/yeoman-generator`

File: .eslintrc.js [hunk 1]
```diff
@@ -1,110 +1,62 @@
 module.exports = {
-	"root": true,
-	"plugins": ["node"],
-	"extends": ["eslint:recommended", "plugin:node/recommended"],
-	"env": {
-		"node": true,
-		"es6": true,
-		"jest": true
+	// "root": true,
```
Comment (evenstensberg) [inline .eslintrc.js]: remove unneeded comments?

File: packages/generate-loader/index.ts [hunk 2]
```diff
@@ -1,8 +1,6 @@
 import LoaderGenerator from "@webpack-cli/generators/loader-generator";
 import * as yeoman from "yeoman-environment";
 
-import { IYeoman } from "./types/Yeoman";
```
Comment (evenstensberg) [inline packages/generate-loader/index.ts]: Why is type removed?

File: .eslintrc.js (hunk 1 above)
Comment (ematipico) [reply .eslintrc.js]: Will do

**Comment by ematipico** [comment]
Yes changes to the functions were entirely to the types, most of them it was to apply the cast. I don't why it was building honestly, there were a number of typescript issues. Anyway, they are all fixed

File: packages/utils/package-manager.ts [hunk 3]
```diff
@@ -109,3 +99,12 @@ export function getPathToGlobalPackages(): string {
 
 	return require("global-modules");
 }
+
+export function spawnChild(pkg: string): SpawnSyncReturns<Buffer> {
```
Comment (anshumanv) [inline packages/utils/package-manager.ts]: Docstrings for this are left above.

File: packages/utils/validate-identifier.ts [hunk 4]
```diff
@@ -79,17 +71,13 @@ function isKeyword(code: string): boolean {
 
 /* tslint:disable: max-line-length */
```
Comment (anshumanv) [inline packages/utils/validate-identifier.ts]: Can be removed.

File: packages/utils/validate-identifier.ts [hunk 5]
```diff
@@ -79,17 +71,13 @@ function isKeyword(code: string): boolean {
...
 /* tslint:enable: max-line-length */
```
Comment (anshumanv) [inline packages/utils/validate-identifier.ts]: Remove

File: .gitignore [hunk 6]
```diff
@@ -37,3 +37,4 @@ yarn.lock
 
 # source maps of docs
 docs/**/*.map
+junit.xml
```
Comment (anshumanv) [inline .gitignore]: thanks

**Pull request merged**